src/app/services/company/company.service.ts

```diff
@@ -35,4 +35,16 @@ export class CompanyService extends BaseService {
         const url = `${environment.backend.url}${Company.URL}/activation/${activationToken}`;
         return this.request(METHOD.POST, url, httpOptions);
     }
+
+    public resetPassword(newPassword: string, resetToken: string): Observable<any> {
+        const httpOptions = { headers: this.headers };
+        const url = `${environment.backend.url}${Company.URL}/reset-password/${resetToken}`;
+        return this.httpClient.post(url, { password: newPassword }, httpOptions);
```

Comment:
Podes usar o método `BaseService.request`. A ideia deste método é centralizar todas as chamadas ao BE de forma a conseguirmos tratar as respostas/erros mais facilmente (no futuro)